keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+- Use CRDs for VolumeSnapshot Namespace Transfer request/approval.
+- Edit external SnapshotController to accept/reject/init VolumeSnapshot transfer.
+- The source and target VolumeSnapshot will point to the same VolumeSnapshotContent.
+- Use Validation Webhook to reject any transfer contains secret.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Transfer of resources other than VolumeSnapshot will not be discussed.
+- Transfer of VolumeSnapshots which contain secret.
+## Proposal
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+`VolumeSnapshotTransferRequest` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: VolumeSnapshotTransferRequest
+metadata:
+  name: transfer-foo
+  namespace: target-namespace
+spec:
+  source:
+    name: foo
+    namespace: source-namespace
+    kind: VolumeSnapshot
+  approvalName: approve-foo
+  targetName: bar
+```
+
+`VolumeSnapshotTransferApproval` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: VolumeSnapshotTransferApproval
+metadata:
+  name: approve-foo
+  namespace: source-namespace
+spec:
+  source:
+    name: foo
+    kind: VolumeSnapshot
+  targetNamespace: target-namespace
+  requestName: transfer-foo
+```
+
+When matching resources are detected, the controller initiate the transfer process.
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+Users should be able to transfer VolumeSnapshots that reference the same VolumeSnapshotContent
+across namespaces. This allows users to quickly share snapshots, utilizing data for testing
+or backups in separate namespaces.
+#### Story 1
+
+- The `prod` namespace contains a PVC that holds user information. VolumeSnapshots are taken of
+this PVC on a regular basis.
+
+- An existing VolumeSnapshot can be transferred to a test namespace. The user initiates the request
+by creating a `VolumeSnapshotTransferRequest` in the source namespace.
+
+- Once created, a `VolumeSnapshotTransferApproval` resource in the target namespace will be created.
+If matching resources are found, the VolumeSnapshot controller can init the transfer. 
+
+- When the transfer is successful, the original VolumeSnapshot should be deleted to assure there is one-to-one mapping between VolumeSnapshot and VolumeSnapshotContent.
```

Comment:
A new VolumeSnapshotContent has to be created and the original one will be deleted when deleting the original VolumeSnapshot.  Need to change the deletion policy for the content to Retain before deleting the VolumeSnapshot.

Reply:
@xing-yang  I understand that we need to change the deletion policy in the original VolumeSnapshot, to avoid deleting the VolumeSnapshotContent mistakenly. However, afaik if we create a new VolumeSnapshotContent, changing the deletion policy is not mandatory, please correct me if I am wrong